cli/src/main.rs

```diff
@@ -0,0 +1,144 @@
+#![recursion_limit = "192"]
+
+#[macro_use]
+extern crate clap;
+#[macro_use]
+extern crate quote;
+#[macro_use]
+extern crate serde_derive;
+
+mod rustfmt;
+
+use std::{collections::BTreeMap, fs, path::Path};
+
+fn read_file(p: &Path) -> String {
+    fs::read_to_string(p).expect(&format!("Error: could not read {}", p.to_str().unwrap()))
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct TomlProject {
+    name: String,
+    version: String,
+    authors: Vec<String>,
+    edition: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct TomlManifest {
+    package: TomlProject,
+    #[serde(serialize_with = "toml::ser::tables_last")]
+    dependencies: BTreeMap<String, toml::Value>,
+}
+
+fn gen_cargo_toml(cargo_path: &Path) -> String {
+    let mut mf: TomlManifest = toml::from_str(&read_file(cargo_path)).unwrap();
+    mf.package.name += "-abi";
+    mf.dependencies = mf
+        .dependencies
+        .into_iter()
+        .filter(|(name, _version)| name.starts_with("owasm-"))
+        .collect();
+    toml::to_string(&mf).unwrap()
+}
+
+fn main() {
+    let matches = app_from_crate!()
+        .arg(
+            clap::Arg::with_name("contract-crate")
+                .index(1)
+                .required(true)
+                .help("Path to contract crate"),
+        )
+        .arg(
+            clap::Arg::with_name("force")
+                .short("f")
+                .help("Overwrite existing ABI crate."),
+        )
+        .get_matches();
+
+    let crate_path = Path::new(matches.value_of("contract-crate").unwrap());
+    let lib_path = crate_path.join("src/lib.rs");
+    let cargo_path = crate_path.join("Cargo.toml");
+
+    let mut abi_crate_path = std::env::current_dir().unwrap();
+    abi_crate_path.push(format!(
```

Comment:
Might be best to generate the crate in the contract directory because the generated JSON and WASM need to be combined for truffle compatibility.